actionpack/lib/action_controller/metal/http_authentication.rb

```diff
@@ -347,7 +347,7 @@ def opaque(secret_key)
     #     private
     #       def authenticate
     #         authenticate_or_request_with_http_token do |token, options|
-    #           token == TOKEN
+    #           ActiveSupport::SecurityUtils.variable_size_secure_compare(token, TOKEN)
```

Comment:
Can you change to `secure_compare`, `variable_size_secure_compare` is an internal method.